FinniversKit/Sources/Fullscreen/FrontPage/FrontPageView.swift

```diff
@@ -269,6 +297,17 @@ public final class FrontPageView: UIView, BasicFrontPageView {
         
     }
     
+    private func addFrontPageShelf() {
+        guard shelfViewModel != nil else { return }
```

Comment:
This is called every time we change `shelfViewModel` from the app. Just in case we do it twice, we could add a `frontPageShelfView?.removeFromSuperview()`, to make it impossible to add multiple.